avocado/utils/iso9660.py

```diff
@@ -293,11 +332,19 @@ def close(self):
 
         :rtype: None
         """
-        if os.path.ismount(self.mnt_dir):
-            process.run('fuser -k %s' % self.mnt_dir, ignore_status=True)
-            process.run('umount %s' % self.mnt_dir)
-
-        shutil.rmtree(self.mnt_dir)
+        if self._mnt_dir:
+            if os.path.ismount(self._mnt_dir):
+                process.run('fuser -k %s' % self.mnt_dir, ignore_status=True)
+                process.run('umount %s' % self.mnt_dir)
```

Comment:
There is no sudo support yet. This PR contains several improvements, I started with defining the API and after that I'm adding the sudo support, which wasn't there before. So this commit focuses only on adding `mnt_dir` support, there is no `sudo` support at all. Later in the PR there is unrelated change which adds the sudo and it treats all necessary places.

So I could reverse the order, but that would mean re-implementing everything as the order matters. On the other hand if I keep the order, it's still consistent as it brings one feature at a time. The sudo was not supported before...